core/app/models/spree/zone.rb

```diff
@@ -58,6 +58,8 @@ def self.match(address)
     # current zone, if it's a state zone. If the passed-in zone has members, it
     # will also be in the result set.
     def self.with_shared_members(zone)
+      return none unless zone
```

Comment:
What John said. However, medium term that entire method goes away when #783 gets applied.